src/Shared/PackageManagers/ProjectManagerFactory.cs

```diff
@@ -8,55 +8,153 @@ namespace Microsoft.CST.OpenSource.PackageManagers
     using System.Linq;
     using System.Net.Http;
 
-    public static class ProjectManagerFactory
+    public class ProjectManagerFactory
     {
         /// <summary>
-        /// Get an appropriate project manager for package given its PackageURL.
+        /// The delegate to use when defining the function to create a project manager with.
+        /// The only runtime parameter we need is the destination directory. Everything else can be defined in the constructor.
         /// </summary>
-        /// <param name="purl">The <see cref="PackageURL"/> for the package to create the project manager for.</param>
-        /// <param name="httpClientFactory"> The <see cref="IHttpClientFactory"/> for the project manager to use for making Http Clients to make web requests.</param>
-        /// <param name="destinationDirectory">The directory to use to store any downloaded packages.</param>
-        /// <returns> BaseProjectManager object </returns>
-        public static BaseProjectManager? CreateProjectManager(PackageURL purl, IHttpClientFactory? httpClientFactory = null, string? destinationDirectory = null)
+        public delegate BaseProjectManager? ConstructProjectManager(string destinationDirectory = ".");
+
+        /// <summary>
+        /// The dictionary of project managers.
+        /// </summary>
+        private readonly Dictionary<string, ConstructProjectManager> _projectManagers;
+
+        /// <summary>
+        /// Initializes a new instance of a <see cref="ProjectManagerFactory"/>.
+        /// </summary>
+        /// <param name="overrideManagers">Use a custom set of managers to override the defaults with.</param>
```

Comment:
```suggestion
        /// <param name="overrideManagers">If provided, will override each matching manager in the defaults - or add if not present in default.</param>
```
Needs to be clear it doesn't override all defaults it only overrides the specified ones. Not sure if this is the best way to write it.